MekHQ/src/mekhq/campaign/unit/Unit.java

```diff
@@ -1675,135 +1678,141 @@ public Money getBuyCost() {
-                +"<scenarioId>"
-                +scenarioId
-                +"</scenarioId>");
-        pw1.println(MekHqXmlUtil.indentStr(indentLvl+1)
-                +"<daysToArrival>"
-                +daysToArrival
-                +"</daysToArrival>");
-        pw1.println(MekHqXmlUtil.indentStr(indentLvl+1)
-                +"<daysSinceMaintenance>"
-                +daysSinceMaintenance
-                +"</daysSinceMaintenance>");
-        pw1.println(MekHqXmlUtil.indentStr(indentLvl+1)
-                +"<daysActivelyMaintained>"
-                +daysActivelyMaintained
-                +"</daysActivelyMaintained>");
-        pw1.println(MekHqXmlUtil.indentStr(indentLvl+1)
-                +"<astechDaysMaintained>"
-                +astechDaysMaintained
-                +"</astechDaysMaintained>");
-        pw1.println(MekHqXmlUtil.indentStr(indentLvl+1)
-                +"<mothballTime>"
-                +mothballTime
-                +"</mothballTime>");
-        pw1.println(MekHqXmlUtil.indentStr(indentLvl+1)
-                +"<mothballed>"
-                +mothballed
-                +"</mothballed>");
-        pw1.println(MekHqXmlUtil.indentStr(indentLvl+1)
-                +"<fluffName>"
-                +MekHqXmlUtil.escape(fluffName)
-                +"</fluffName>");
-        pw1.println(MekHqXmlUtil.indentStr(indentLvl+1)
-                +"<history>"
-                +MekHqXmlUtil.escape(history)
-                +"</history>");
-        if (null != refit) {
-            refit.writeToXml(pw1, indentLvl+1);
+        if (salvaged) {
+            MekHqXmlUtil.writeSimpleXmlTag(pw1, indent, "salvaged", true);
+        }
+
+        if (site != SITE_BAY) {
+            MekHqXmlUtil.writeSimpleXmlTag(pw1, indent, "site", site);
+        }
+
+        if (forceId != Force.FORCE_NONE) {
+            MekHqXmlUtil.writeSimpleXmlTag(pw1, indent, "forceId", forceId);
+        }
+
+        if (scenarioId != Scenario.S_DEFAULT_ID) {
+            MekHqXmlUtil.writeSimpleXmlTag(pw1, indent, "scenarioId", scenarioId);
+        }
+
+        if (daysToArrival > 0) {
+            MekHqXmlUtil.writeSimpleXmlTag(pw1, indent, "daysToArrival", daysToArrival);
+        }
+
+        if (daysSinceMaintenance > 0) {
+            MekHqXmlUtil.writeSimpleXmlTag(pw1, indent, "daysSinceMaintenance", daysSinceMaintenance);
+        }
+
+        if (daysActivelyMaintained > 0) {
+            MekHqXmlUtil.writeSimpleXmlTag(pw1, indent, "daysActivelyMaintained", daysActivelyMaintained);
+        }
+
+        if (astechDaysMaintained > 0) {
+            MekHqXmlUtil.writeSimpleXmlTag(pw1, indent, "astechDaysMaintained", astechDaysMaintained);
+        }
+
+        if (mothballTime > 0) {
+            MekHqXmlUtil.writeSimpleXmlTag(pw1, indent, "mothballTime", mothballTime);
+        }
+
+        if (mothballed) {
+            MekHqXmlUtil.writeSimpleXmlTag(pw1, indent, "mothballed", true);
+        }
+
+        if (!fluffName.isEmpty()) {
```

Comment:
Can this be null?

Reply:
Good catch. I've pushed a fix where the fluff name can't be null but is now only an empty string.